src/pip/_internal/metadata/base.py

```diff
@@ -263,3 +281,27 @@ def iter_installed_distributions(
         if user_only:
             it = (d for d in it if d.in_usersite)
         return (d for d in it if d.canonical_name not in skip)
+
+
+class Wheel(Protocol):
+    location: str
+
+    def as_zipfile(self) -> zipfile.ZipFile:
+        raise NotImplementedError()
+
+
+class FilesystemWheel(Wheel):
```

Comment:
nit: `FileSystemWheel`?

Reply:
The usage between file system and filesystem seem quite split and I don’t really have a preference.